src/domain_tests/TestRunner.cs

```diff
@@ -0,0 +1,1103 @@
+                    {
+                        [System.Serializable]
+                        public class Cls 
+                        {
+                            public static event Action Before;
+                            public static void Call()
+                            {
+                                Before();
+                            }
+                        }
+                    }",
+                DotNetAfter = @"
+                    using System;
+                    namespace TestNamespace
+                    {
+                        [System.Serializable]
+                        public class Cls
+                        {
+                            public static event Action After;
+                            public static void Call()
+                            {
+                                After();
+                            }
+                        }
+                    }",
+                PythonCode = @"
+import clr
+import sys
+clr.AddReference('DomainTests')
+from TestNamespace import Cls
+
+called = False
+
+def callback_function():
+    global called
+    called = True
+
+def before_reload():
+    global called
+    called = False
+    Cls.Before += callback_function
+    Cls.Call()
+    assert called is True
+
+def after_reload():
+    global called
+    assert called is True
+    called = False
+    Cls.Call()
+    assert called is False
+    #try:
+    #    assert 2 == Cls.Before
+    #except TypeError:
+    #    print('Caught expected exception')
+    #else:
+    #    raise AssertionError('Failed to throw exception')
+",
+            },
+
+            new TestCase
+            {
+                Name = "namespace_rename",
+                DotNetBefore = @"
+                    namespace TestNamespace
+                    {
+                        [System.Serializable]
+                        public class Cls 
+                        {
+                            public int Foo;
+                            public Cls(int i)
+                            {
+                                Foo = i;
+                            }
+                        }
+                    }",
+                DotNetAfter = @"
+                    namespace NewTestNamespace
+                    {
+                        [System.Serializable]
+                        public class Cls 
+                        {
+                            public int Foo;
+                            public Cls(int i)
+                            {
+                                Foo = i;
+                            }
+                        }
+                    }",
+                PythonCode = @"
+import clr
+import sys
+clr.AddReference('DomainTests')
+import TestNamespace
+
+def before_reload():
+    sys.my_cls = TestNamespace.Cls
+    sys.my_inst = TestNamespace.Cls(1)
+
+def after_reload():
+     try:
+        TestNamespace.Cls(2)
+        sys.my_cls.Member()
```

Comment:
It'd be easier to isolate the regression with two separate try/except/else here.